docs/pages/docs/guide/advanced/typescript.mdx

```diff
@@ -1,7 +1,97 @@
 # TypeScript
 
-import { Callout } from 'nextra/components'
+Nextra is built with TypeScript and provides excellent TypeScript support out of
+the box. This guide will help you leverage TypeScript in your Nextra project.
 
-<Callout emoji="🚨">
-  This page is a stub. Help us expand it by contributing!
-</Callout>
+## Getting Started
+
+To use TypeScript in your Nextra project, you need to:
+
+1. Install TypeScript and related dependencies:
+
+```bash
+npm install typescript @types/react @types/node
+```
+
+2. Create a `tsconfig.json` file in the root of your project:
```

Comment:
we don't really need to manually create this file, it will automatically created